mosaic-runtime/src/commonTest/kotlin/com/jakewharton/mosaic/stuff.kt

```diff
@@ -48,6 +48,18 @@ internal val MosaicNode.position: IntOffset
 
 class Holder<T>(var value: T)
 
+internal fun MosaicNode.nodeByTestTag(tag: String): MosaicNode {
+	val queue = mutableListOf(this@nodeByTestTag)
+	while (queue.isNotEmpty()) {
+		val node = queue.removeAt(0)
```

Comment:
"deque" == double-ended queue, i.e., efficient appends and removes from both ends.
```suggestion
	val queue = ArrayDeque().also { it.add(this) }
	while (queue.isNotEmpty()) {
		val node = queue.removeFirst()
```

Reply:
This might fail to build because of lack of generics, but hoping K2 can figure it out.

Reply:
Yep. Failed.

https://youtrack.jetbrains.com/issue/KT-40346